packages/network-controller/src/NetworkController.ts

```diff
@@ -1374,6 +1374,10 @@ export class NetworkController extends BaseController<
    * @returns networkClientId of the network configuration with the given chainId
    */
   findNetworkClientIdByChainId(chainId: Hex): NetworkClientId {
+    if (this.state.providerConfig.chainId === chainId) {
+      return this.state.selectedNetworkClientId;
+    }
```

Comment:
Hmmm this may not work.  Seems like they can be out of sync.

![image](https://github.com/MetaMask/core/assets/3500406/2114caa5-4f89-4ee1-86cd-35cf77fe9ed7)